lite2/client.go

```diff
@@ -454,6 +459,45 @@ func (c *Client) TrustedValidatorSet(height int64, now time.Time) (*types.Valida
 	return c.trustedStore.ValidatorSet(height)
 }
 
+// updateMissingTrustedHeaderAndVals finds the previous trusted height and then uses the sequence method to verify
+// all the headers at each height up to the inputted height.
+// If all are verified updates the store and returns the header and validator set.
+func (c *Client) updateMissingTrustedHeaderAndVals(height int64) (*types.SignedHeader, *types.ValidatorSet, error) {
+	var (
+		prevTrustedHeight int64
+		newHeader         *types.SignedHeader
+		newVals           *types.ValidatorSet
+		err               error
+	)
+	prevTrustedHeight, err = c.previousTrustedHeight(height)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.trustedHeader, err = c.trustedStore.SignedHeader(prevTrustedHeight)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.trustedNextVals, err = c.trustedStore.ValidatorSet(prevTrustedHeight)
+	if err != nil {
+		return nil, nil, err
+	}
+	newHeader, err = c.primary.SignedHeader(height)
```

Comment:
we have a function for that - `c.fetchHeaderAndValsAtHeight(height)`

Reply:
immediately after we should check if it's not expired I think. before checking if we can trust it

Reply:
Checking if it is expired is done in the `Verify` function which is part of the `sequence` functions